anta/tests/security.py

```diff
@@ -377,3 +378,73 @@ def test(self) -> None:
                 failed_log = f"SSL certificate `{certificate.certificate_name}` is not configured properly:"
                 failed_log += get_failed_logs(expected_certificate_details, actual_certificate_details)
                 self.result.is_failure(f"{failed_log}\n")
+
+
+class VerifyIpv4ACL(AntaTest):
+    """
+    This class verifies the configuration and the correct operation of the IPv4 access lists.
+
+    Expected results:
+        * success: The test will pass if an IPv4 ACL is configured with the correct sequence entries.
+        * failure: The test will fail if an IPv4 ACL is not configured or entries are not in sequence.
+    """
+
+    name = "VerifyIpv4ACL"
+    description = "Verifies the configuration and the correct operation of the IPv4 access lists."
+    categories = ["security"]
+    commands = [AntaTemplate(template="show ip access-lists {acl}")]
+
+    class Input(AntaTest.Input):
+        """Inputs for the VerifyIpv4ACL test."""
+
+        ipv4_access_list: List[Ipv4Acl]
+        """List of IPv4 ACL to verify"""
+
+        class Ipv4Acl(BaseModel):
+            """Detail of IPv4 ACl"""
+
+            name: str
+            """Name of IPv4 ACL"""
+
+            entries: List[Ipv4AclEntries]
+            """List of IPv4 ACL entries"""
+
+            class Ipv4AclEntries(BaseModel):
+                """IPv4 ACL entries details"""
+
+                sequence: int
+                """Sequence number of ACL entry"""
+                action: str
+                """Action of an ACL entry"""
+
+    def render(self, template: AntaTemplate) -> list[AntaCommand]:
+        return [template.render(acl=acl.name, entries=acl.entries) for acl in self.inputs.ipv4_access_list]
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        self.result.is_success()
+        for command_output in self.instance_commands:
+            # Collecting input ACL details
+            acl_name = command_output.params["acl"]
+            acl_entries = command_output.params["entries"]
+
+            # Check if ACL is configured
+            ipv4_acl_list = command_output.json_output["aclList"]
+            if not ipv4_acl_list:
+                self.result.is_failure(f"{acl_name}: Not found")
```

Comment:
I would prefer a single failure message: The following IPv4 ACL(s) are not configured: ACL_TEST1, ACL_TEST2, etc.